_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,206 @@
+
+The rule applies to `iframe` elements within a given [document context](#document-context) and [exposed to assistive technologies](#exposed-to-assistive-technologies) that have a [non-empty](#non-empty) accessible name.
+
+**Note:** 
+- Nested `iframe` contents treat the parent `iframe` element as the [document context](#document-context).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is unique within the [document context](#document-context).
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [http://www.w3.org/TR/WCAG20-TECHS/H64.html](http://www.w3.org/TR/WCAG20-TECHS/H64.html)
+- [https://www.w3.org/TR/2008/REC-WCAG20-20081211/#ensure-compat-rsv](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#ensure-compat-rsv)
+
+## Test cases
+
+### Passed
+
+#### Pass example 1
+
+Usage of `title` attribute to describe the `iframe` content, and there is only one iframe with in document context.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+```
+
+#### Pass example 2
+
+Multiple `iframe` elements in the document having different `title` descriptions as accessible name.
+
+```html
+<iframe title="List of Contributors to Repository 1" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors to Repository 2" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Pass example 3
+
+Multiple `iframe` elements in the document having different `aria-label` descriptions as accessible name.
+
+```html
+<iframe aria-label="List of Contributors to Repository 1" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors to Repository 2" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Pass example 4
+
+Multiple `iframe` elements in the document having different `aria-labelledby` descriptions as accessible name.
+
+```html
+<div id="desc-for-title">List of Contributors</div>
+<iframe aria-labelledby="desc-for-title" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<div id="desc-for-title1">List of Reviewers</div>
+<iframe aria-labelledby="desc-for-title1" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Pass example 5
+
+`frames` having the same `title` within a given document context, but one of them is not exposed to assistive technologies.
+
+```html
+<iframe aria-hidden="true" title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+
+#### Pass example 5
+
+`frames` are allowed to have the same `title` across different document contexts.
+
+```html
```

Comment:
Is that nested frame supposed to represent the document inside the outer frame? This example isn't clear.

Reply:
Yes, that is the thinking. I have expanded the description to add more clarity.

Reply:
It might be clearer to expand `iframe#level2-frame1` to `iframe` with `id` `level2-frame1`
I would also end the sentence with "and does not share the document context of `iframe` with `id` `level1-frame1`"

Reply:
This test case doesn't express what its supposed to. The body of an iframe element is used for iframe fallbacks, not for "nested frame content". Also, because the src is the same in the frames with the same title, this wouldn't fail even if they were in the same page.